### PR TITLE
Remove glusterfs fuse client mount library version check for `auto_unmount` mount option.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by jsafrane** [comment]
@k8s-bot ok to test

File: pkg/volume/glusterfs/glusterfs.go [hunk 1]
```diff
@@ -435,6 +389,24 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 			glog.Infof("glusterfs: successfully mounted %s", dir)
 			return nil
 		}
+
+		// Give a try without `auto_unmount mount option, because
+		// it could be that gluster fuse client is older version and
+		// mount.glusterfs is unaware of `auto_unmount`.
+		// Use a mount string without `auto_unmount``
+		autoMountOptions := mountOptions
+		for i := len(autoMountOptions) - 1; i >= 0; i-- {
```
Comment (jsafrane) [inline pkg/volume/glusterfs/glusterfs.go]: I stared at this loop for quite a long time to see what it does and why it needs to process autoMountOptions backwards... IMO, straightforward approach would be more readable and only a bit less efficient (it depends of nr. of auto_unmounts in the array):

```go
autoMountOptions := make([]string, len(mountOptions))
for _, opt := range mountOptions {
    if opt != "auto_unmount" {
         autoMountOptions = append(autoMountOptions, opt)
    }
}
```

Comment (humblec) [inline pkg/volume/glusterfs/glusterfs.go]: True. Its just about the 'efficiency'. I do agree more readable approach is as shown above, let me convert to that. Thanks !

**Comment by humblec** [comment]
@jsafrane  @rootfs The plan was to add `auto_unmount` option from glusterfs fuse  v3.11, however considering the requests/asks  for this option in earlier versions of glusterfs, community has decided to backport it to older versions. So, the version check has to be replaced with another approach as in this patch.

**Pull request merged**